truffleruby/src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -649,7 +646,7 @@ public Object evalBadBinding(DynamicObject source, DynamicObject badBinding, Not
             // TODO CS 14-Apr-15 concat space + code as a rope, otherwise the string will be copied after the rope is converted
             final Source source = Source.newBuilder(space + RopeOperations.decodeRope(code)).name(filename).mimeType(RubyLanguage.MIME_TYPE).build();
 
-            final MaterializedFrame frame = Layouts.BINDING.getFrame(binding);
+            final MaterializedFrame frame = BindingNodes.getExtrasFrame(getContext(), binding);
```

Comment:
Should we always create an extra frame here? Or is the top frame in the binding enough?